packages/p2p/src/components/modal-manager/modal-manager-context-provider.jsx

```diff
@@ -11,6 +11,44 @@ const ModalManagerContextProvider = props => {
     const [is_modal_open, setIsModalOpen] = React.useState(false);
     const [modal_props, setModalProps] = React.useState(new Map());
     const { general_store } = useStores();
+    const persisted_states = React.useRef({});
+
+    /**
+     * A `useState` wrapper that allows the local state to be persisted and restored if the modal is unmounted in place of another modal.
+     * By default the local states are not saved when modal is unmounted using hideModal, but once `hideModal` is called with the setting `should_restore_local_state` to `true`,
+     * local states will be saved and automatically restored when the modal is mounted back.
+     *
+     * @param {key} string - the key to specify when persisting the local state, by default you should specify the local state name
+     * @param {default_state} - the value you want the state to be initially
+     */
+    const useSavedState = (key, default_state) => {
+        const [saved_state, setSavedState] = React.useState(default_state);
+        const saved_state_ref = React.useRef(saved_state);
+
+        React.useEffect(() => {
+            const persisted_state = persisted_states.current[active_modal.key];
+
+            if (persisted_state) {
+                if (persisted_state[key]) {
+                    setSavedState(persisted_state[key]);
+                }
+            } else {
+                persisted_states.current[active_modal.key] = {
+                    [key]: default_state,
+                };
+            }
+
+            return () => {
+                persisted_states.current[active_modal.key][key] = saved_state_ref.current;
+            };
+        }, []);
+
+        React.useEffect(() => {
+            saved_state_ref.current = saved_state;
+        }, [saved_state]);
+
+        return [saved_state, setSavedState];
+    };
```

Comment:
This hook will allow modals that has their own local state to be persisted when it is unmounted, depending if `should_restore_local_state` is set to `true` in `hideModal` settings.